indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -158,7 +160,9 @@ public boolean run()
 
       JobHelper.injectSystemProperties(job);
 
-      if (config.isCombineText()) {
+      if(config.getInputFormatClass() != null) {
```

Comment:
this is done at multiple places, prob add a util method to JobHelper to JobHelper.setinputFormat(job, config) and have the if checks at single place.